test/unit/core/window/test-history.js

```diff
@@ -327,6 +327,7 @@ describes.sandboxed('Window - History', {}, (env) => {
     });
 
     it('should create natural binding and make it singleton', () => {
+      win.AMP_CONFIG = {test: false};
```

Comment:
Why is it preferred to stub the test environment this way versus stubbing `isTest` or `isProd` fns to return as needed?

Reply:
How can we stub `isTest`? AFAIK it isn't possible without abusing the module system (i.e. stubbing a namespace import since we compile to cjs w/ babel).

We could expose a fn for `setIsTestForTesting` with `@visibleForTesting`....but figured I would follow the convention of what we are already doing for now.

Reply:
rip, that's basically what I did in #35669: https://github.com/caroqliu/amphtml/blob/1846a584e84ecdf803c9a6050e543528cba59703/extensions/amp-ad/0.1/test/test-amp-ad-3p-impl.js#L331

do you mind explaining more why this is bad/leaky? and also is it worth disallowing this type of testing pattern to prevent people from doing it in the future?